protobuf/lib/src/protobuf/field_set.dart

```diff
@@ -280,6 +280,38 @@ class _FieldSet {
     _setNonExtensionFieldUnchecked(meta, fi, value);
   }
 
+  /// Sets a non-repeated nullable field with error-checking.
+  ///
+  /// Works for both extended and non-extended fields.
+  /// Suitable for public API.
+  void _setFieldNullable(int tagNumber, Object? value) {
+    final meta = _meta;
+    var fi = _nonExtensionInfo(meta, tagNumber);
+    if (fi == null) {
+      final extensions = _extensions;
+      if (extensions == null) {
+        throw ArgumentError('tag $tagNumber not defined in $_messageName');
+      }
+      if (value == null) {
+        _clearField(tagNumber);
+        return;
+      }
+      extensions._setField(tagNumber, value);
+      return;
+    }
+
+    if (fi.isRepeated) {
+      throw ArgumentError(_setFieldFailedMessage(
+          fi, value, 'repeating field (use get + .add())'));
+    }
+    if (value == null) {
+      _clearField(tagNumber);
```

Comment:
This will remove `required` fields from the field set, right? I think that's probably OK (removing required fields is already allowed with `clearField`), but the documentation says "sets a non-repeated nullable field", and my understanding is only optional fields are nullable.

I think we need two documentation, one for this member with implementation details, one for the `GeneratedMessage.setFieldNullable` with how the member is supposed to be used. This should say it will just clear the field (same as `_clearField`). `GeneratedMessage.setFieldNullable` should clarify what happens when you pass `null`.

Reply:
Indeed, I hadn't noticed that!
I'll add some documentation.

Reply:
I just pushed the changes in the documention, I'm not sure if it's right. :smile: